src/main/java/frc/robot/util/AutoLogic.java

```diff
@@ -0,0 +1,248 @@
+
+    tab.addString("Current AutoElevatorCommand(s)", () -> autoAction.getSelected())
+        .withPosition(3, 1)
+        .withSize(2, 1);
+
+    tab.add("Tested Autos", testedAutos)
+        .withWidget(BuiltInWidgets.kComboBoxChooser)
+        .withPosition(3, 0)
+        .withSize(2, 1);
+
+    autoAction.setDefaultOption("EXTEND AND WAIT", "raiseElevatorandWait");
+    autoAction.addOption("EXTEND WHILE MOVING", "raiseElevator");
+  }
+
+  public static <T> String[] toStringArray(List<T> dataList) {
+
+    String[] data = new String[dataList.size()];
+
+    for (int i = 0; i < dataList.size(); i++) {
+
+      String addedData = dataList.get(i).toString();
+
+      data[i] = "\n" + addedData;
+    }
+
+    return data;
+  }
+
+  public static PathPlannerTrajectory makeTrajectory(
+      PathPlannerPath path,
+      ChassisSpeeds startingSpeeds,
+      Rotation2d startingRotation,
+      RobotConfig config) {
+
+    PathPlannerTrajectory trajectory =
+        new PathPlannerTrajectory(path, startingSpeeds, startingRotation, config);
+
+    return trajectory;
+  }
+
+  public static void addAutoOptions() {
+    autoPicker.setDefaultOption("DEFAULT PATH", "PreLoadAuto");
+    autoPicker.addOption("Mid 3 Piece", "Mid3Piece");
+    autoPicker.addOption("Low 3 Piece", "Low3Piece");
+    autoPicker.addOption("DRIVE ONLY", "SideWall");
+    autoPicker.addOption("Low  1 L4", "OppsidetoD");
+    autoPicker.addOption("Upper 3 Piece", "Upper3Piece");
+    autoPicker.addOption("CHOREO CRAZY TEST?(IGNORE)", "4 L4 Coral");
+    autoPicker.addOption("CHOREO 3 PIECE?(IGNORE)", "New Choreo");
+    autoPicker.addOption("CHOREO 3 PIECE LESS CRAZY?(IGNORE)", "Triple7");
+    autoPicker.addOption("High 1 L4", "SideToJ");
+    autoPicker.addOption("CHOREO High 4 L4(IGNORE)", "HighL4");
+    autoPicker.addOption("TESTING Low 3 Piece", "TEST Low3Piece");
+    autoPicker.addOption("ALIGNMENT TEST", "ALIGN TEST");
+    autoPicker.addOption("DRIVE TEST", "DRIVE1FT");
+  }
+
+  public static void addTestedAutos() {}
+
+  public static Command lowerElevator() {
+    if (r.superStructure != null) {
+      return r.superStructure.intake();
+    }
+    return Commands.none().withName("lowerElevator");
+  }
+
+  public static Command raiseElevator() {
+    if (r.superStructure != null) {
+      return r.superStructure.levelFour(() -> true);
+    }
+    return Commands.none().withName("raiseElevator");
+  }
+
+  public static Command lowerElevatorandWait() {
+
+    return Commands.sequence(new WaitCommand(0.8), lowerElevator())
+        .withName("lowerElevatorandWAIT");
+  }
+
+  public static Command raiseElevatorandWait() {
+    return Commands.sequence(new WaitCommand(0.8), raiseElevator())
+        .withName("raiseElevatorandWAIT");
+  }
+
+  public static Command autoAlignmentCommand() {
+    return AutoAlign.autoAlign(s.drivebaseSubsystem).withName("AutoAlign");
+  }
+
+  public static void registerCommand() {
+
+    if (RobotState.isDisabled()) {
```

Comment:
It is possible for the robot to be enabled while running the code constructors. That will happen if the rio reboots (browns out) mid-match as the WPILib code can get the 'enabled' signal before the various class / Java initializers run.
So, it's not safe to run this only if the robot is initialized - some other method should be used to ensure these are only registered once (if that's the intention) or in certain states.